network.cpp

```diff
@@ -64,24 +64,52 @@ bool operator!=(const network& lhs, const network& rhs)
 
 network change_all_weights(network n, double new_weight)
 {
-    for(auto& layer : n.get_net_weights())
-        for(auto& node : layer)
-            for(auto& weight : node)
```

Comment:
I like more this form and the way this was working before.
Just copy the weights and then use this range loop.